### PR TITLE
Regen bindings for upcoming resync

### PR DESCRIPTION
**Pull request by Manishearth**
<s>The StyleComplexColor stuff has broken things, so the sync may take longer to fix.

cc @xidorn if you want to look into this. Otherwise I'll fix it monday.</s>

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13539)

### CODE REVIEW

**Comment by Manishearth** [comment]
Managed to fix it, testing now

**Comment by Manishearth** [comment]
Works now, wikipedia runs.

r? @bholley or @emilio

**Comment by Manishearth** [comment]
Trypush at https://treeherder.mozilla.org/#/jobs?repo=try&revision=9951d19cdf83, will be forwarded to s-i tip if it works.

**Comment by Manishearth** [comment]
Try passed, merged with incubator. Any fixups here can be picked up in the next sync.

File: components/style/gecko_bindings/sugar/style_complex_color.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            Color::RGBA(rgba) => convert_rgba_to_nscolor(&rgba).into(),
+            Color::CurrentColor => StyleComplexColor::current_color(),
+        }
+    }
+}
+
+impl From<StyleComplexColor> for Color {
+    fn from(other: StyleComplexColor) -> Self {
+        if other.mForegroundRatio == 0 {
+            return Color::RGBA(convert_nscolor_to_rgba(other.mColor))
```
Comment (emilio) [inline components/style/gecko_bindings/sugar/style_complex_color.rs]: nit: ditch the `return` keywords, or the extra condition chains.

File: components/style/properties/gecko.mako.rs [hunk 2]
```diff
@@ -240,71 +240,40 @@ def set_gecko_property(ffi_name, expr):
...
-            },
-            Color::RGBA(rgba) => convert_rgba_to_nscolor(&rgba),
-        };
+        % if complex_color:
+            let result = v.into();
+        % else:
+            use cssparser::Color;
+            let result = match v {
+                Color::RGBA(rgba) => convert_rgba_to_nscolor(&rgba),
+                Color::CurrentColor => 0,
```
Comment (emilio) [inline components/style/properties/gecko.mako.rs]: Where are we using this, can/should this be represented on Servo in any other way? If it's illegal to specify currentcolor for these properties, we should reject it for parsing (I think we don't right now), and mark this branch as `unreachable!`. At least an issue open for that referenced here would be great.

File: components/style/gecko_bindings/sugar/style_complex_color.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        } else if other.mForegroundRatio == 255 {
+            return Color::CurrentColor
+        } else {
+            // FIXME handle interpolation values
```
Comment (emilio) [inline components/style/gecko_bindings/sugar/style_complex_color.rs]: Open an issue and reference it from here?

**Review by emilio — APPROVED** [review]
The rest of the changes look fine. r=me with those.

File: components/style/properties/gecko.mako.rs (hunk 2 above)
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: They can all be currentColor, I think that the code hasn't been updated yet to use complex colors for all of them (@heycam?)

**Comment by upsuper** [comment]
@Manishearth FWIW, my GitHub ID is @upsuper. @xidorn is not controlled by me.

**Pull request merged**